### DISCUSSION TITLE
How to fetch a secret file secret or decrypt an exist file?

### DISCUSSION THREAD

**Discussion by rignaneseleo** [codemagic-ci-cd/codemagic-docs#1638] [Q&A]
Hi, I've got a game to deploy that is on a public repo on GitHub.

There a file in the repo that needs to be decrypted with https://github.com/brunocampos01/encrypt-file before the app runs.

Is there a way to do so? Or is there a way to import this secret file directly in codemagic?

**Comment by dtrdic** [comment]
Hey Leo, yes, you can encrypt it with encrypt-file. After that, you can decrypt it in Pre-build script section, just save the password in an environment variable, you can check more about it [here](https://docs.codemagic.io/yaml-basic-configuration/configuring-environment-variables/)

**Comment by rignaneseleo** [comment]
I solved using this code:

```
#!/bin/bash

for folder in opentabu/assets/words/*/; do
  if [[ -f "${folder}words.csv.enc" ]]; then
    openssl enc -aes-256-cbc -d -salt -pass pass:PW -in "${folder}words.csv.enc" -out "${folder}words.csv"
  fi
done
```
